### PR TITLE
Finish merge of #320 to edge

### PR DESCRIPTION
**Pull request by giseburt**
Since I had to rebase the work of @benlaurie onto edge, and merge that in with other changes, there is a chance I've broken stuff. This should be fairly easy to test.

Once it has been tested, we can merge this in.

### CODE REVIEW

File: g2core/stepper.cpp [hunk 1]
```diff
@@ -1042,6 +1042,31 @@ stat_t st_get_pwr(nvObj_t *nv)
...
+
+    Motors[motor]->setStepPolarity((ioMode)nv->value);
+    return (STAT_OK);
+}
+
+stat_t st_get_sp(nvObj_t *nv)            // get motor step polarity
+{
+    if (nv->value < 0) { return (STAT_INPUT_LESS_THAN_MIN_VALUE); }
+    if (nv->value > 1) { return (STAT_INPUT_EXCEEDS_MAX_VALUE); }
+
```
Comment (benlaurie) [inline g2core/stepper.cpp]: Seems like this would allow values like .5, with I know not what consequences.

File: g2core/stepper.cpp [hunk 2]
```diff
@@ -1122,7 +1147,8 @@ static const char fmt_0tr[] = "[%s%s] m%s travel per revolution%10.4f%s\n";
 static const char fmt_0mi[] = "[%s%s] m%s microsteps%16d [1,2,4,8,16,32]\n";
 static const char fmt_0su[] = "[%s%s] m%s steps per unit %17.5f steps per%s\n";
 static const char fmt_0po[] = "[%s%s] m%s polarity%18d [0=normal,1=reverse]\n";
-static const char fmt_0ep[] = "[%s%s] m%s enable polarity%11d [0=active HIGH,1=ractive LOW]\n";
+static const char fmt_0ep[] = "[%s%s] m%s enable polarity%11d [0=active HIGH,1=active LOW]\n";
+static const char fmt_0sp[] = "[%s%s] m%s step polarity%13d [0=active HIGH,1=active LOW]\n";
```
Comment (benlaurie) [inline g2core/stepper.cpp:1141]: 0 for active high and 1 for active low seems perverse.

File: g2core/stepper.h [hunk 3]
```diff
@@ -337,7 +337,7 @@ typedef enum {
  *  There are 5 main structures involved in stepper operations;
  *
  *  data structure:                   found in:      runs primarily at:
- *    mpBuffer planning buffers (bf)    planner.c      main loop
+ *    mpBuffer planning buffers (bf)    planner.c       main loop
```
Comment (benlaurie) [inline g2core/stepper.h:340]: why change alignment?

**Comment by benlaurie** [comment]
Apart from comments, LGTM.

Note: can't test at the moment.

File: g2core/stepper.cpp (hunk 2 above)
Comment (giseburt) [reply g2core/stepper.cpp:1141]: Indeed. The fix for that is coming in a future PR.

File: g2core/stepper.h (hunk 3 above)
Comment (giseburt) [reply g2core/stepper.h:340]: I suspect this wasn't intentional, but I don't recall.

File: g2core/stepper.cpp (hunk 1 above)
Comment (giseburt) [reply g2core/stepper.cpp]: Hmm, good point. @aldenhart do you feel we should do more here?

**Comment by giseburt** [comment]
Now we have a merge to complete, but we should pull this one in next.

**Comment by aldenhart** [comment]
Merged with edge 101.02 and added back in Enable Polarity and Step Polarity with all JSON and text mode support. 

Note that the sense of the polarity settings are in flux. Should we use the current sense (0 means active lo) or the new sense that will be in effect with the GPIO enhancements, which are imminent?  The description of which is [here](https://github.com/synthetos/g2/wiki/gpio-primitives#pon-polarity-setting). 

I believe that this code uses the current sense, and reports in JSON correctly, but note that the text in text mode is reversed. But I have not gotten to an oscilloscope to get ground truth yet.

**Pull request merged**